From 7237f3e611eb5bdfe5604a89fcfdf22e3a27532c Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@2eb42a87-3827-0410-b767-dc15055982e4>
Date: Wed, 21 Mar 2012 15:34:20 +0000
Subject: [PATCH 079/462] Mantis 22213: Build: remove exhausted components do
 not link libraries to libraries fixed: dependency to system libraries
 (advapi32.lib)

git-svn-id: svn://localhost/SelfServiceBuild/trunk@107 2eb42a87-3827-0410-b767-dc15055982e4
---
 .../Build/cpp/makefile.common.mak             | 12 +++---
 SelfServiceBuild/Build/cpp/makefile.rules.mak | 39 ++++++++-----------
 2 files changed, 24 insertions(+), 27 deletions(-)

diff --git a/SelfServiceBuild/Build/cpp/makefile.common.mak b/SelfServiceBuild/Build/cpp/makefile.common.mak
index 5592465..ce27f38 100644
--- a/SelfServiceBuild/Build/cpp/makefile.common.mak
+++ b/SelfServiceBuild/Build/cpp/makefile.common.mak
@@ -266,7 +266,6 @@ LIB_DST_APPSTATSS         = $(_LIB)\appstatss.lib
 LIB_DST_BILLING           = $(_LIB)\billing.lib
 LIB_DST_BILLINGS          = $(_LIB)\billings.lib
 LIB_DST_CARDSCAN          = $(_LIB)\cardscan.lib
-LIB_DST_CFGCLNT           = $(_LIB)\cfgclnt.lib
 LIB_DST_CXAPI             = $(_LIB)\cxapi.lib
 LIB_DST_MAILERS           = $(_LIB)\mailers.lib
 LIB_DST_MAILER            = $(_LIB)\mailer.lib
@@ -311,7 +310,7 @@ LIB_AEAPARSER             = $(LIB_DST_AEAPARSER)
 LIB_ASSO                  = $(LIB_DST_ASSO)
 LIB_GRAPHIC               = $(LIB_DST_GRAPHIC)
 LIB_TEXT                  = $(LIB_DST_TEXT) $(LIB_BOOST)
-LIB_SYSTOOLS              = $(LIB_DST_SYSTOOLS) $(SYS_ADVAPI32)
+LIB_SYSTOOLS              = $(LIB_DST_SYSTOOLS)
 LIB_TOOLS                 = $(LIB_DST_TOOLS)
 LIB_ALERTS                = $(LIB_DST_ALERTS)
 LIB_APPSTATS              = $(LIB_DST_APPSTATS)
@@ -319,7 +318,6 @@ LIB_APPSTATSS             = $(LIB_DST_APPSTATSS)
 LIB_BILLING               = $(LIB_DST_BILLING)
 LIB_BILLINGS              = $(LIB_DST_BILLINGS)
 LIB_CARDSCAN              = $(LIB_DST_CARDSCAN)
-LIB_CFGCLNT               = $(LIB_DST_CFGCLNT)
 LIB_CXAPI                 = $(LIB_DST_CXAPI)
 LIB_MAILERS               = $(LIB_DST_MAILERS)
 LIB_MAILER                = $(LIB_DST_MAILER)
@@ -328,8 +326,8 @@ LIB_MASSAIS               = $(LIB_DST_MASSAIS)
 LIB_MCFGCLNT              = $(LIB_DST_MCFGCLNT) $(LIB_MSYSTOOLS)
 LIB_MCFGCLNTS             = $(LIB_DST_MCFGCLNTS) $(LIB_MSYSTOOLS)
 LIB_MCFGACCESS            = $(LIB_DST_MCFGACCESS)
-LIB_MCORBA                = $(LIB_DST_MCORBA) $(LIB_MSYSTOOLS)
-LIB_MCORBAS               = $(LIB_DST_MCORBAS) $(LIB_MSYSTOOLS)
+LIB_MCORBA                = $(LIB_DST_MCORBA) $(LIB_MSYSTOOLS) $(LIB_MCFGCLNT)
+LIB_MCORBAS               = $(LIB_DST_MCORBAS) $(LIB_MSYSTOOLS) $(LIB_MCFGCLNTS)
 LIB_MDSCPRINTER           = $(LIB_DST_MDSCPRINTER)
 LIB_MTOOLS                = $(LIB_DST_MTOOLS)
 LIB_TRACES                = $(LIB_DST_TRACES) $(LIB_MCFGCLNTS) $(LIB_MCORBAS) $(LIB_MASSAIS) $(SYS_ACETAO_S)
@@ -404,6 +402,8 @@ SYS_WINSPOOL     = winspool.lib
 SYS_WS2_32       = ws2_32.lib
 SYS_WSOCK32      = wsock32.lib
 
+COMMON_LINKLIBS  = $(SYS_ADVAPI32)
+
 ######################################################################
 # SOURCE-GENERATION
 ######################################################################
@@ -412,3 +412,5 @@ GEN_MASSAI_COMMON= $(COMMONROOT)\Massai\idl\Massai
 GEN_MASSAI_CUSS=   $(CUSSROOT)\Massai\idl\massai
 GEN_CUSS=          $(CUSSROOT)\Massai\idl\cuss
 GEN_MASSAI_ABLS=   $(ABLSROOT)\Massai\idl\Massai
+
+
diff --git a/SelfServiceBuild/Build/cpp/makefile.rules.mak b/SelfServiceBuild/Build/cpp/makefile.rules.mak
index 739c8bb..940059c 100644
--- a/SelfServiceBuild/Build/cpp/makefile.rules.mak
+++ b/SelfServiceBuild/Build/cpp/makefile.rules.mak
@@ -438,7 +438,7 @@ $(MY_DLL1): $(_OBJ) $(MY_DLL1_OBJS) $(MY_DLL1_RES) $(MY_DLL1_USER_LIBS)
     @echo ------- Linking $@
     @if exist $@ echo del $@ && del $@
     $(LD) $(STDDLLFLAGS) $(MY_LDFLAGS) $(MY_DLL1_LDFLAGS) $(MY_DLL1_USER_LIBS) \
-        $(MY_DLL1_LINKLIBS) $(MY_DLL1_OBJS) $(MY_DLL1_RES) \
+        $(MY_DLL1_LINKLIBS) $(COMMON_LINKLIBS) $(MY_DLL1_OBJS) $(MY_DLL1_RES) \
         -implib:$(_LIB)\$(@B).lib \
         -map:$(_DLL)\$(@B).map \
         -pdb:$(_DLL)\$(@B).pdb \
@@ -456,7 +456,7 @@ $(MY_DLL2): $(_OBJ) $(MY_DLL2_OBJS) $(MY_DLL2_RES) $(MY_DLL2_USER_LIBS)
     @echo ------- Linking $@
     @if exist $@ echo del $@ && del $@
     $(LD) $(STDDLLFLAGS) $(MY_LDFLAGS) $(MY_DLL2_LDFLAGS) $(MY_DLL2_USER_LIBS) \
-        $(MY_DLL2_LINKLIBS) $(MY_DLL2_OBJS) $(MY_DLL2_RES) \
+        $(MY_DLL2_LINKLIBS) $(COMMON_LINKLIBS) $(MY_DLL2_OBJS) $(MY_DLL2_RES) \
         -implib:$(_LIB)\$(@B).lib \
         -map:$(_DLL)\$(@B).map \
         -pdb:$(_DLL)\$(@B).pdb \
@@ -480,7 +480,7 @@ $(MY_BIN1): \
     @echo ------- Linking $@
     @if exist $@ echo del $@ && del $@
     $(LD) $(STDBINFLAGS) $(MY_LDFLAGS) $(MY_BIN1_LDFLAGS) $(MY_BIN1_USER_LIBS) \
-        $(MY_BIN1_LINKLIBS) $(MY_BIN1_OBJS) $(MY_BIN1_RES) \
+        $(MY_BIN1_LINKLIBS) $(COMMON_LINKLIBS) $(MY_BIN1_OBJS) $(MY_BIN1_RES) \
         -implib:$(_OBJ)\$(@B).lib \
         -out:$@
     @echo ------- Linking done.
@@ -499,7 +499,7 @@ $(MY_BIN2): \
     @echo ------- Linking $@
     @if exist $@ echo del $@ && del $@
     $(LD) $(STDBINFLAGS) $(MY_LDFLAGS) $(MY_BIN2_LDFLAGS) $(MY_BIN2_USER_LIBS) \
-        $(MY_BIN2_LINKLIBS) $(MY_BIN2_OBJS) $(MY_BIN2_RES) \
+        $(MY_BIN2_LINKLIBS) $(COMMON_LINKLIBS) $(MY_BIN2_OBJS) $(MY_BIN2_RES) \
         -implib:$(_OBJ)\$(@B).lib \
         -out:$(MY_BIN2)
     @echo ------- Linking done.
@@ -517,7 +517,7 @@ $(MY_BIN3): \
     @echo ------- Linking $@
     @if exist $@ echo del $@ && del $@
     $(LD) $(STDBINFLAGS) $(MY_LDFLAGS) $(MY_BIN3_LDFLAGS) $(MY_BIN3_USER_LIBS) \
-        $(MY_BIN3_LINKLIBS) $(MY_BIN3_OBJS) $(MY_BIN3_RES) \
+        $(MY_BIN3_LINKLIBS) $(COMMON_LINKLIBS) $(MY_BIN3_OBJS) $(MY_BIN3_RES) \
         -implib:$(_OBJ)\$(@B).lib \
         -out:$(MY_BIN3)
     @echo ------- Linking done.
@@ -535,7 +535,7 @@ $(MY_BIN4): \
     @echo ------- Linking $@
     @if exist $@ echo del $@ && del $@
     $(LD) $(STDBINFLAGS) $(MY_LDFLAGS) $(MY_BIN4_LDFLAGS) $(MY_BIN4_USER_LIBS) \
-        $(MY_BIN4_LINKLIBS) $(MY_BIN4_OBJS) $(MY_BIN4_RES) \
+        $(MY_BIN4_LINKLIBS) $(COMMON_LINKLIBS) $(MY_BIN4_OBJS) $(MY_BIN4_RES) \
         -implib:$(_OBJ)\$(@B).lib \
         -out:$(MY_BIN4)
     @echo ------- Linking done.
@@ -553,7 +553,7 @@ $(MY_BIN5): \
     @echo ------- Linking $@
     @if exist $@ echo del $@ && del $@
     $(LD) $(STDBINFLAGS) $(MY_LDFLAGS) $(MY_BIN5_LDFLAGS) $(MY_BIN5_USER_LIBS) \
-        $(MY_BIN5_LINKLIBS) $(MY_BIN5_OBJS) $(MY_BIN5_RES) \
+        $(MY_BIN5_LINKLIBS) $(COMMON_LINKLIBS) $(MY_BIN5_OBJS) $(MY_BIN5_RES) \
         -implib:$(_OBJ)\$(@B).lib \
         -out:$(MY_BIN5)
     @echo ------- Linking done.
@@ -571,7 +571,7 @@ $(MY_BIN6): \
     @echo ------- Linking $@
     @if exist $@ echo del $@ && del $@
     $(LD) $(STDBINFLAGS) $(MY_LDFLAGS) $(MY_BIN6_LDFLAGS) $(MY_BIN6_USER_LIBS) \
-        $(MY_BIN6_LINKLIBS) $(MY_BIN6_OBJS) $(MY_BIN6_RES) \
+        $(MY_BIN6_LINKLIBS) $(COMMON_LINKLIBS) $(MY_BIN6_OBJS) $(MY_BIN6_RES) \
         -implib:$(_OBJ)\$(@B).lib \
         -out:$(MY_BIN6)
     @echo ------- Linking done.
@@ -589,7 +589,7 @@ $(MY_BIN7): \
     @echo ------- Linking $@
     @if exist $@ echo del $@ && del $@
     $(LD) $(STDBINFLAGS) $(MY_LDFLAGS) $(MY_BIN7_LDFLAGS) $(MY_BIN7_USER_LIBS) \
-        $(MY_BIN7_LINKLIBS) $(MY_BIN7_OBJS) $(MY_BIN7_RES) \
+        $(MY_BIN7_LINKLIBS) $(COMMON_LINKLIBS) $(MY_BIN7_OBJS) $(MY_BIN7_RES) \
         -implib:$(_OBJ)\$(@B).lib \
         -out:$(MY_BIN7)
     @echo ------- Linking done.
@@ -607,7 +607,7 @@ $(MY_BIN8): \
     @echo ------- Linking $@
     @if exist $@ echo del $@ && del $@
     $(LD) $(STDBINFLAGS) $(MY_LDFLAGS) $(MY_BIN8_LDFLAGS) $(MY_BIN8_USER_LIBS) \
-        $(MY_BIN8_LINKLIBS) $(MY_BIN8_OBJS) $(MY_BIN8_RES) \
+        $(MY_BIN8_LINKLIBS) $(COMMON_LINKLIBS) $(MY_BIN8_OBJS) $(MY_BIN8_RES) \
         -implib:$(_OBJ)\$(@B).lib \
         -out:$(MY_BIN8)
     @echo ------- Linking done.
@@ -625,7 +625,7 @@ $(MY_BIN9): \
     @echo ------- Linking $@
     @if exist $@ echo del $@ && del $@
     $(LD) $(STDBINFLAGS) $(MY_LDFLAGS) $(MY_BIN9_LDFLAGS) $(MY_BIN9_USER_LIBS) \
-        $(MY_BIN9_LINKLIBS) $(MY_BIN9_OBJS) $(MY_BIN9_RES) \
+        $(MY_BIN9_LINKLIBS) $(COMMON_LINKLIBS) $(MY_BIN9_OBJS) $(MY_BIN9_RES) \
         -implib:$(_OBJ)\$(@B).lib \
         -out:$(MY_BIN9)
     @echo ------- Linking done.
@@ -643,7 +643,7 @@ $(MY_BIN10): \
     @echo ------- Linking $@
     @if exist $@ echo del $@ && del $@
     $(LD) $(STDBINFLAGS) $(MY_LDFLAGS) $(MY_BIN10_LDFLAGS) $(MY_BIN10_USER_LIBS) \
-        $(MY_BIN10_LINKLIBS) $(MY_BIN10_OBJS) $(MY_BIN10_RES) \
+        $(MY_BIN10_LINKLIBS) $(COMMON_LINKLIBS) $(MY_BIN10_OBJS) $(MY_BIN10_RES) \
         -implib:$(_OBJ)\$(@B).lib \
         -out:$(MY_BIN10)
     @echo ------- Linking done.
@@ -661,7 +661,7 @@ $(MY_BIN11): \
     @echo ------- Linking $@
     @if exist $@ echo del $@ && del $@
     $(LD) $(STDBINFLAGS) $(MY_LDFLAGS) $(MY_BIN11_LDFLAGS) $(MY_BIN11_USER_LIBS) \
-        $(MY_BIN11_LINKLIBS) $(MY_BIN11_OBJS) $(MY_BIN11_RES) \
+        $(MY_BIN11_LINKLIBS) $(COMMON_LINKLIBS) $(MY_BIN11_OBJS) $(MY_BIN11_RES) \
         -implib:$(_OBJ)\$(@B).lib \
         -out:$(MY_BIN11)
     @echo ------- Linking done.
@@ -679,7 +679,7 @@ $(MY_BIN12): \
     @echo ------- Linking $@
     @if exist $@ echo del $@ && del $@
     $(LD) $(STDBINFLAGS) $(MY_LDFLAGS) $(MY_BIN12_LDFLAGS) $(MY_BIN12_USER_LIBS) \
-        $(MY_BIN12_LINKLIBS) $(MY_BIN12_OBJS) $(MY_BIN12_RES) \
+        $(MY_BIN12_LINKLIBS) $(COMMON_LINKLIBS) $(MY_BIN12_OBJS) $(MY_BIN12_RES) \
         -implib:$(_OBJ)\$(@B).lib \
         -out:$(MY_BIN12)
     @echo ------- Linking done.
@@ -697,7 +697,7 @@ $(MY_BIN13): \
     @echo ------- Linking $@
     @if exist $@ echo del $@ && del $@
     $(LD) $(STDBINFLAGS) $(MY_LDFLAGS) $(MY_BIN13_LDFLAGS) $(MY_BIN13_USER_LIBS) \
-        $(MY_BIN13_LINKLIBS) $(MY_BIN13_OBJS) $(MY_BIN13_RES) \
+        $(MY_BIN13_LINKLIBS) $(COMMON_LINKLIBS) $(MY_BIN13_OBJS) $(MY_BIN13_RES) \
         -implib:$(_OBJ)\$(@B).lib \
         -out:$(MY_BIN13)
     @echo ------- Linking done.
@@ -715,7 +715,7 @@ $(MY_BIN14): \
     @echo ------- Linking $@
     @if exist $@ echo del $@ && del $@
     $(LD) $(STDBINFLAGS) $(MY_LDFLAGS) $(MY_BIN14_LDFLAGS) $(MY_BIN14_USER_LIBS) \
-        $(MY_BIN14_LINKLIBS) $(MY_BIN14_OBJS) $(MY_BIN14_RES) \
+        $(MY_BIN14_LINKLIBS) $(COMMON_LINKLIBS) $(MY_BIN14_OBJS) $(MY_BIN14_RES) \
         -implib:$(_OBJ)\$(@B).lib \
         -out:$(MY_BIN14)
     @echo ------- Linking done.
@@ -733,7 +733,7 @@ $(MY_BIN15): \
     @echo ------- Linking $@
     @if exist $@ echo del $@ && del $@
     $(LD) $(STDBINFLAGS) $(MY_LDFLAGS) $(MY_BIN15_LDFLAGS) $(MY_BIN15_USER_LIBS) \
-        $(MY_BIN15_LINKLIBS) $(MY_BIN15_OBJS) $(MY_BIN15_RES) \
+        $(MY_BIN15_LINKLIBS) $(COMMON_LINKLIBS) $(MY_BIN15_OBJS) $(MY_BIN15_RES) \
         -implib:$(_OBJ)\$(@B).lib \
         -out:$(MY_BIN15)
     @echo ------- Linking done.
@@ -884,11 +884,6 @@ clean::
 !endif
 
 
-######################################################################
-# System Libraries
-######################################################################
-
-$(SYS_ADVAPI32):
 
 ######################################################################
 # Dependencies
-- 
2.41.0.windows.1

